proto/src/determined/experiment/v1/searcher.proto

```diff
@@ -143,9 +149,14 @@ message SetSearcherProgressOperation {
 message CompleteValidateAfterOperation {
   // The ValidateAfterOperation being completed.
   ValidateAfterOperation op = 1;
-  // The value of searcher metric associated with this completed metric.
-  // The metric provided should be the metric used to guide HP search.
-  double searcher_metric = 2;
+  // A searcher metric or all metrics.
+  oneof metrics {
+    // The value of searcher metric associated with this completed metric.
+    // The metric provided should be the metric used to guide HP search.
+    double searcher_metric = 2;
+    // All validation metrics.
+    google.protobuf.Struct all_metrics = 3;
+  }
```

Comment:
Why offer two separate types rather than one union type?

In this case, a google.protobuf.Struct for both, I suppose, since that is the union type.

Reply:
It is not really the union type, is it? [The doc](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#struct) says (and that is what I observe as well, or am I missing something?) that it is a protobuf equivalent of a JSON map.

Reply:
oh, I assumed that Struct was equivalent of a json element, able to take on any valid json type.

Reply:
Oh, why not use [this](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Value) then?

Reply:
Hmm. Maybe. Although, the definition of `Value` is: